### PR TITLE
feat: Add `--web-config-file` flag to the config reloader

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

ref: #5932
Add the new flag to the config-reloader binary

```
--web-config-file=<string> 
        [EXPERIMENTAL] Path to configuration file that can enable TLS or authentication. See: https://prometheus.io/docs/prometheus/latest/configuration/https/
```

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
--web-config-file=<string> 
        [EXPERIMENTAL] Path to configuration file that can enable TLS or authentication. See: https://prometheus.io/docs/prometheus/latest/configuration/https/
```

### CODE REVIEW

File: cmd/prometheus-config-reloader/main.go [hunk 1]
```diff
@@ -76,6 +77,11 @@ func main() {
 		"address on which to expose metrics (disabled when empty)").
 		String()
 
+	webConfig := app.Flag(
+		"web-config-file",
+		"Path to configuration file that can enable TLS or authentication. See: https://prometheus.io/docs/prometheus/latest/configuration/https/",
```
Comment (nicolastakashi) [inline cmd/prometheus-config-reloader/main.go:82]: I'm wondering if worth to mention this is experimental, since Prometheus is warning about this as well.

Comment (dongjiang1989) [reply cmd/prometheus-config-reloader/main.go:82]: Thanks you for you review. Experimental warning added.

File: cmd/prometheus-config-reloader/main.go [hunk 2]
```diff
@@ -91,6 +97,12 @@ func main() {
 
 	versionutil.RegisterIntoKingpinFlags(app)
 
+	err := web.Validate(*webConfig)
```
Comment (simonpasquier) [inline cmd/prometheus-config-reloader/main.go:100]: hmm this should happen after? ideally after the logging has been setup.

File: cmd/prometheus-config-reloader/main.go [hunk 3]
```diff
@@ -158,11 +170,15 @@ func main() {
...
-		srv := http.Server{Addr: *listenAddress}
+		srv := &http.Server{}
 
 		g.Add(func() error {
 			level.Info(logger).Log("msg", "Starting web server for metrics", "listen", *listenAddress)
-			return srv.ListenAndServe()
+			return web.ListenAndServe(srv, &web.FlagConfig{
+				WebListenAddresses: &[]string{*listenAddress},
+				WebConfigFile:      webConfig,
+				WebSystemdSocket:   nil,
```
Comment (simonpasquier) [inline cmd/prometheus-config-reloader/main.go:180]:
```suggestion
```

File: cmd/prometheus-config-reloader/main.go (hunk 2 above)
Comment (dongjiang1989) [reply cmd/prometheus-config-reloader/main.go]: Thanks for your review. 
Done. Move after the logging setuped

File: cmd/prometheus-config-reloader/main.go [hunk 4]
```diff
@@ -106,6 +112,12 @@ func main() {
 		stdlog.Fatal(err)
 	}
 
+	err = web.Validate(*webConfig)
+	if err != nil {
+		level.Error(logger).Log("msg", "Unable to validate web configuration file. err: "+err.Error())
```
Comment (simonpasquier) [inline cmd/prometheus-config-reloader/main.go:117]: (nit) better to have a separate key for the error
```suggestion
		level.Error(logger).Log("msg", "Unable to validate web configuration file", "err", err)
```

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Pull request merged**